cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -265,7 +265,10 @@ private <T> KubeTwillController monitorController(KubeLiveInfo liveInfo, long ti
               liveInfo.getApplicationName(), runId, timeout, timeoutUnit);
 
     // Schedule to terminate the controller in the timeout time.
-    Future<?> terminationFuture = monitorScheduler.schedule(controller::terminate, timeout, timeoutUnit);
+    Future<?> terminationFuture = monitorScheduler.schedule(() -> {
+      controller.setJobTimedout(true);
```

Comment:
k8s job fails to start could happen if scheduling of job doesnt happen?

Reply:
Yes I think that's possible. The failure mode that I tested was job failing to start because of permissions issues.